packages/enzyme/src/ShallowWrapper.js

```diff
@@ -170,6 +171,13 @@ class ShallowWrapper {
       privateSet(this, RENDERER, renderer);
       this[RENDERER].render(nodes, options.context);
       const { instance } = this[RENDERER].getNode();
+      const adapter = getAdapter(this[OPTIONS]);
+      const lifecycles = getAdapterLifecycles(adapter);
+      // Ensure to call componentDidUpdate when instance.setState is called
+      if (instance && lifecycles.componentDidUpdate.onSetState && !instance[SET_STATE]) {
+        privateSet(instance, SET_STATE, instance.setState);
+        instance.setState = (...args) => this.setState(...args);
```

Comment:
With this change i am now seeing alot of failing tests which have  a setState in componentDidMount. `Method "setState" is only meant to be run on a single node. undefined found instead.`

Reply:
See #1756.